vcert/connection_cloud.py

```diff
@@ -607,22 +622,122 @@ def set_policy(self, zone, policy_spec):
         else:
             # Application does not exist. Create one
-            app_req = build_app_create_request(app_name, user_details, resp_cit_data)
+            app_req = build_app_create_request(app_name, owners_list, resp_cit_data)
             status, data = self._post(URLS.APPLICATIONS, app_req)
             if status != HTTPStatus.CREATED:
                 raise VenafiError(f"Could not create application [{app_name}]")
         return
 
+    def resolve_owners(self, users_list, user_details):
+        owners_list = list()
+        for_update = False
+        if not users_list:
+            # When no users are provided on the list, adds apikey user as owner
+            api_owner = resolve_apikey_owner(user_details)
+            owners_list.append(api_owner)
+            for_update = True
+        else:
+            # Resolving the usernames list
+            # Creating a higher level Teams object to cache the response
+            teams_list = list()
+            users_list = list(set(users_list))
+            for user in users_list:
+                cloud_owner = self.resolve_user_to_cloud_owner(user)
+                if cloud_owner:
+                    owners_list.append(cloud_owner)
+                else:
+                    if not teams_list:
+                        status, response = self._get(URLS.TEAMS_ID)
+                        data_team = response['teams']
+                        for t in data_team:
+                            team = build_team(t)
+                            teams_list.append(team)
+                    team_owner = self.resolve_user_to_cloud_team(teams_list, user)
+                    if not team_owner:
+                        raise VenafiError(f"Unable to find team [{user}]")
+                    owners_list.append(team_owner)
+        return for_update, owners_list
+
+    def resolve_user_to_cloud_owner(self, username):
+        user = self.get_vaas_identity(username)
+        if not user:
+            return None
+        owner = create_owner(OWNER_TYPE_USER, user.user_id)
+        return owner
+
+    def resolve_user_to_cloud_team(self, teams, username):
+        if not teams:
+            try:
+                status, response = self._get(URLS.TEAMS_ID)
+                data = response['teams']
+                teams = build_team(data)
+            except VenafiError:
+                log.debug(f"Error while getting team [{username}]")
+        for team in teams:
+            if team.name == username:
+                owner = create_owner(OWNER_TYPE_TEAM, team.team_id)
+                return owner
+        return None
+
+    def get_vaas_identity(self, username):
+        if not username or username == "":
+            raise VenafiError("Username cannot be empty")
+        url = URLS.USERS_USERNAME.format(username)
+        try:
+            status, response = self._get(url)
+            identities = response['users'][0]
```

Comment:
The response assignation should happen after the status has been checked. Move line 707 below the "if" conditional